sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ConvertHelpers.java

```diff
@@ -86,10 +85,11 @@ public static <T> ConvertedSchemaInformation<T> getConvertedSchemaInformation(
       // output.
       convertedSchema =
           new ConvertedSchemaInformation<>((SchemaCoder<T>) SchemaCoder.of(inputSchema), null);
-    } else if (outputType.equals(TypeDescriptor.of(GenericRecord.class))) {
-      convertedSchema =
-          new ConvertedSchemaInformation<T>(
-              (SchemaCoder<T>) AvroUtils.schemaCoder(AvroUtils.toAvroSchema(inputSchema)), null);
+      //    } else if (outputType.equals(TypeDescriptor.of(GenericRecord.class))) {
```

Comment:
I temporary commented this "if" case but I'm stuck on how to properly handle this without adding a dependency on Avro extension or copying the needed code and keeps Avro dependency.

@mosche Do you have any idea on this?

Reply:
@aromanenko-dev The only possible way I see to decouple this is to use an AutoService with a provider and a provider registrar as done in many different places, e.g. SchemaProvider and SchemaProviderRegistrar.